shed/event_streams.py

```diff
@@ -676,34 +682,37 @@ class filter(EventStream):
 
     def __init__(self, predicate, child, *args, input_info,
                  full_event=False, document_name='event', **kwargs):
-        """Initialize the node
-        """
+        """Initialize the node """
         self.predicate = predicate
 
         EventStream.__init__(self, child, input_info=input_info, **kwargs)
-        self.kwargs = kwargs
-        self.args = args
+        for k in self.pop_kwargs:
+            if k in kwargs:
+                kwargs.pop(k)
+
+        self.func_kwargs = kwargs
+        self.func_args = args
         self.full_event = full_event
         self.document_name = document_name
-        if document_name == 'event':
-            self.event = self._event
-        else:
-            self.start = self._start
+        self.tv = None
```

Comment:
I would perform cleanup upon a start (for reasons that @tacaswell mentioned on the callback side) since we an do more debugging. I'm not certain we do need to clean this up though. As we always will produce a new tv (truth value). But I can put it in if you want.